plonky2x/src/frontend/builder/mux.rs

```diff
@@ -0,0 +1,186 @@
+use plonky2::field::extension::Extendable;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::target::BoolTarget;
+
+use crate::prelude::{CircuitBuilder, CircuitVariable, Variable};
+
+impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
+    pub fn select_index_with_select<T: CircuitVariable>(
+        &mut self,
+        selector: Variable,
+        inputs: &[T],
+    ) -> T {
+        // Serialized variables for the circuit variable.
+        let num_var_in_t = inputs[0].variables().len();
+        let input_len = inputs.len();
+
+        let mut res = inputs[0].variables();
+
+        for i in 0..input_len {
+            let target_i = self.constant::<Variable>(F::from_canonical_usize(i));
+            let diff = self.sub(target_i, selector);
```

Comment:
This is a bit inefficient in terms of rows. 


A better way to implement it would be as an inner product. In particular, using builder.api.add_many can sum a bunch of Targets in one row. There might also be a method for an inner product, not sure.

Reply:
i push a manual inner product is it what u suggest? run with vec size 10 but not really improved
![image](https://github.com/succinctlabs/succinctx/assets/104515688/16f7713e-67f4-4bc5-936f-3760b46e6e1c)